Rosetta 2.0 documentation (#2372)
dfinity/portal: docs/developer-docs/defi/rosetta/icp_rosetta/data_api/network.md

> # How to fetch network information
> For most endpoints you will require some information about the network represented as a [NetworkIdentifier](https://www.rosetta-api.org/docs/models/NetworkIdentifier.html).
> For this part of the guide we assume your rosetta instance is up and running under the address `0.0.0.0:8081`.
> 
> ## Fetch Network List
> The network list [endpoint](https://www.rosetta-api.org/docs/NetworkApi.html#networklist) will give you information about the NetworkIdentifier you need to use for ICP Rosetta. It requires no arguments and can thus also be used as a health check of ICP rosetta. You can fetch the endpoint using the following command:  
> 

Reviewer: ```suggestion
---
keywords: [rosetta, icp, data api, rosetta api, defi]
---

import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";

# Fetch network information

<MarkdownChipRow labels={["Intermediate", "Tutorial"  ]} />

## Overview

For most endpoints you will require some information about the network represented as a [NetworkIdentifier](https://www.rosetta-api.org/docs/models/NetworkIdentifier.html).

### Prerequisites

- Your Rosetta instance is up and running under the address `0.0.0.0:8081`.

## Fetching the network list

The network list [endpoint](https://www.rosetta-api.org/docs/NetworkApi.html#networklist) will give you information about the `NetworkIdentifier` you need to use for ICP Rosetta. It requires no arguments and can thus also be used as a health check of ICP Rosetta. You can fetch the endpoint using the following command:  

```

Reviewer: This file needs to be `.mdx`

```bash
curl --location '0.0.0.0:8081/network/list' --header 'Content-Type: application/json' --data '{
    "metadata": {}
}'
```

> The response will look something like this:

Reviewer: ```suggestion
The response will resemble the following:
```

```bash
{
    "network_identifiers": [
        {
            "blockchain": "Internet Computer",
            "network": "00000000000000020101"
        }
    ]
}
```

> You can use this NetworkIdentifer with other endpoints which will require you to provide it in the data section of your `HTTP` call. 
> 
> ## Fetch Network Options
> This [endpoint](https://www.rosetta-api.org/docs/NetworkApi.html#networkoptions) returns the version information and allowed network-specific types for a NetworkIdentifier.

Reviewer: ```suggestion
You can use this `NetworkIdentifer` with other endpoints which will require you to provide it in the data section of your `HTTP` call. 

## Fetch network options
This [endpoint](https://www.rosetta-api.org/docs/NetworkApi.html#networkoptions) returns the version information and allowed network-specific types for a `NetworkIdentifier`.
```

            {
                "code": 720,
                "message": "Transaction not in the mempool",
                "retriable": false
            },
            {
                "code": 721,
                "message": "Blockchain is empty",
                "retriable": false
            },
            {
                "code": 730,
                "message": "An invalid transaction has been detected",
                "retriable": false
            },
            {
                "code": 740,
                "message": "Internet Computer error",
                "retriable": false
            },
            {
                "code": 750,
                "message": "Transaction rejected",
                "retriable": false
            },
            {
                "code": 770,
                "message": "Operation failed",
                "retriable": true
            },
            {
                "code": 760,
                "message": "Transaction expired",
                "retriable": false
            }
        ],
        "historical_balance_lookup": true,
        "call_methods": [],
        "balance_exemptions": [],
        "mempool_coins": false
    }
}

```

> ## Fetch Network Status

Reviewer: ```suggestion
## Fetch network status
```
